src/snowflake/snowpark/_internal/compiler/large_query_breakdown.py

```diff
@@ -377,8 +403,23 @@ def _replace_child_and_update_ancestors(
                 # Skip if the node is already updated.
                 continue
 
+            valid_to_breakdown, score = self._is_node_valid_to_breakdown(node)
+            if valid_to_breakdown and score >= candidate_score:
+                # Find the valid node with the highest complexity score.
+                candidate_node = node
+                candidate_score = score
+
             update_resolvable_node(node, self._query_generator)
             updated_nodes.add(node)
 
             parents = self._parent_map[node]
             nodes_to_reset.extend(parents)
+
+        if candidate_node is not None:
+            # Assertion for type check. we must have initialized the priority queue
+            # by the time we reach here.
+            assert self.priority_queue is not None
+            # Update the priority queue with the new eligible node in the ancestors.
+            heapq.heappush(
```

Comment:
it doesn't seem we adjust the queue its-self, what happens if the same candidate node occurs in the queue twice? the new score should be the correct score used, but if we did not pop out the old candidate, we might still use the old score for comparison
for example if I have (100, node1), and then node 1 scoe is updated to 50 after breakdown, and it is valid, we will have both (100, node1), (50, node1) in the heap, right?

Reply:
do we have a test for such case also?